fast setup for output tensor in tensor iterator

glaringlee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#32736 fast setup for output tensor in tensor iterator**

Differential Revision: [D19610704](https://our.internmc.facebook.com/intern/diff/D19610704)

Adding fast setup support for two more cases besides contiguous:
1. channels last contiguous tensors.
2. non-contiguous but having exactly same shape/stride, no overlapping and dense tensors.

This supported quantized tensor as well.

Commit:
fast setup for output tensor in tensor iterator

[ghstack-poisoned]

Commit:
Update on "fast setup for output tensor in tensor iterator"


Differential Revision: [D19610704](https://our.internmc.facebook.com/intern/diff/D19610704)

Adding fast setup support for two more cases besides contiguous:
1. channels last contiguous tensors.
2. non-contiguous but having exactly same shape/stride, no overlapping and dense tensors.

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit bf950c4a:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=bf950c4ad01b1001b3003aa21acfa1a5218c382d).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4371979) pytorch_macos_10_13_py3_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=92750802) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=92750802))
<details>
<summary>
<code>Jan 29 12:57:38 RuntimeError: test_autograd failed!</code>
</summary>

```
Jan 29 12:57:37 Ran 1046 tests in 287.676s 
Jan 29 12:57:37  
Jan 29 12:57:37 FAILED (errors=1, skipped=13, expected failures=1) 
Jan 29 12:57:37  
Jan 29 12:57:37 Generating XML reports... 
Jan 29 12:57:38 Traceback (most recent call last): 
Jan 29 12:57:38   File "test/run_test.py", line 456, in <module> 
Jan 29 12:57:38     main() 
Jan 29 12:57:38   File "test/run_test.py", line 449, in main 
Jan 29 12:57:38     raise RuntimeError(message) 
Jan 29 12:57:38 RuntimeError: test_autograd failed! 
Jan 29 12:57:38 + cleanup 
Jan 29 12:57:38 + retcode=1 
Jan 29 12:57:38 + set +x 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

VitalyFedyunin commented:
Please add benchmark numbers for:
1) Old contiguous cases (to check if there are any redundancy).
2) New channels-last-ish cases to proof the is speedups.

VitalyFedyunin commented:
Please add benchmark numbers for:
1) Old contiguous cases (to check if there are any redundancy).
2) New channels-last-ish cases to proof the is speedups.

Commit:
Update on "fast setup for output tensor in tensor iterator"


Differential Revision: [D19610704](https://our.internmc.facebook.com/intern/diff/D19610704)

Adding fast setup support for two more cases besides contiguous:
1. channels last contiguous tensors.
2. non-contiguous but having exactly same shape/stride, no overlapping and dense tensors.

[ghstack-poisoned]

glaringlee commented:
This is too old to merge correctly. Created a new PR
https://github.com/pytorch/pytorch/pull/33165
